dhall/src/Dhall/Map.hs

```diff
@@ -75,8 +75,13 @@ instance (Eq k, Eq v) => Eq (Map k v) where
   Map m1 ks == Map m2 ks' = m1 == m2 && ks == ks'
   {-# INLINABLE (==) #-}
 
+{-|
+>>> fromList [("A",1),("B",2)] < fromList [("B",1),("A",0)]
+True
+-}
 instance (Ord k, Ord v) => Ord (Map k v) where
-  compare (Map mL ksL) (Map mR ksR) = compare mL mR <> compare ksL ksR
+  compare m1 m2 = compare (toList m1) (toList m2)
```

Comment:
I think you could accomplish the same thing more efficiently by a slight modification of the original code:

```suggestion
  compare m1 m2 = compare ksL ksR <> compare mL mR
```

Reply:
I don't believe so. Take

```
fromList [('B',1),('A',2)] < fromList [('B',2),('A',1)]
```

That's `True` with my version, but `False` with yours.

Reply:
Ah, you're right.  Never mind, then